### PR TITLE
Support for `edit-an` command

### LINKED ISSUE #14: As a user, I want to be able to edit a patient's appointment note

### PR DESCRIPTION
**Pull request by sebin0817**
Resolves #14

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditNoteCommand.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        requireNonNull(model);
+        List<Person> lastShownPatientList = model.getFilteredPersonList();
+
+        if (patientIndex.getZeroBased() >= lastShownPatientList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+        } else {
+            if (noteIndex.getZeroBased() >= lastShownPatientList.get(patientIndex.getZeroBased()).getNotes().size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+            }
+        }
```
Comment (joenus) [inline src/main/java/seedu/address/logic/commands/EditNoteCommand.java:80]: Nitpick, but I think we can avoid the additional nesting by place the `if` statement on the top level.

**Review by joenus — CHANGES_REQUESTED** [review]
A few minor issues, but otherwise lgtm!

**Review by joenus — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/EditNoteCommand.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    @Override
+    public String getMessageUsage() {
+        return MESSAGE_USAGE;
+    }
+
+    /**
+     * Creates and returns a {@code Note} with the details of {@code noteToEdit}
+     * edited with {@code editNoteDescriptor}.
+     */
+    private static Note createEditedNote(Note noteToEdit, EditNoteDescriptor editNoteDescriptor) {
```
Comment (jmsandiegoo) [inline src/main/java/seedu/address/logic/commands/EditNoteCommand.java:110]: Good job following the usage of edit command descriptor. Maybe if you have time you could also see how Person builder can be leveraged into notes via notes builder? But not really a must in this case, as descriptor would suffice.

**Review by jmsandiegoo — APPROVED** [review]
LGTM 👍

**Pull request merged**